crates/sui/src/upgrade_compatibility.rs

```diff
@@ -589,23 +593,11 @@ fn compare_packages(
             .get_module_by_name_from_root(name.as_str())
             .context("Unable to get module")?;
 
-        if !file_set.contains(&compiled_unit_with_source.source_path) {
-            let file_contents: Arc<str> =
-                fs::read_to_string(&compiled_unit_with_source.source_path)
-                    .context("Unable to read source file")?
-                    .into();
-            let file_hash = FileHash::new(&file_contents);
-
-            files.insert(
-                file_hash,
-                (
-                    FileName::from(compiled_unit_with_source.source_path.to_string_lossy()),
-                    file_contents,
-                ),
-            );
-
-            file_set.insert(&compiled_unit_with_source.source_path);
-        }
+        insert_into_files(
+            &mut file_set,
+            &mut files,
+            &compiled_unit_with_source.source_path,
```

Comment:
Under what conditions is this file not in the fileset already? 

Would it make more sense to just up front make a file set from all of the source paths + the toml? I'm not sure we will notice a perf difference versus what we are doing here ,and that feels a bit easier to follow than what is going on here.

Reply:
done